### DISCUSSION TITLE
Allow server function to return ReadableStream

### DISCUSSION THREAD

**Discussion by lcswillems** [vercel/next.js#49358] [Ideas]
### Goals

I want to stream data from server to client, and would like to use a ReadableStream for this.

### Background

Don't know background.

### Proposal

Allow server function to return ReadableStream.

**Comment by FraserMuir** [comment]
To put some context around this (as I believe I am facing the same issue) we're looking to be able to return ReadableStream objects from Server Actions. I believe this is an impossibility with the current architecture due to how the return values are serialized from Server Actions. I get the following at runtime when attempting to do exactly that:

![image](https://user-images.githubusercontent.com/25723984/236822890-edc1ab7c-becb-401c-b27f-34c6fc26e15a.png)

It would be good to understand if this is just a temporary limitation that could be solved with some tweaks / additions, or if it's fundamentally impossible. Either way, updating the docs to make this limitation more visible would be ideal: https://nextjs.org/docs/app/building-your-application/data-fetching/server-actions

**Comment by aleksa-codes** [comment]
I am facing this right now, any updates? I am trying to pass stream.toReadableStream(); using OpenAI from server action to the client.

**Comment by yudistiraashadi** [comment]
I believe this is possible since Vercel's AI SDK has stream from server function [link](https://sdk.vercel.ai/docs/ai-sdk-rsc/streaming-values#createstreamablevalue)
